lab-taylor/app/html/index.html

```diff
@@ -0,0 +1,31 @@
+<!DOCTYPE html>
+<html ng-app="cowsayApp">
+  <head>
+    <meta charset="utf-8">
+    <title>__cowsay-client__</title>
+    <script src="bundle.js"></script>
+    <link rel="stylesheet" type="text/css" href="bundle.css" />
+  </head>
+  <body>
+    <main ng-controller="CowsayController">
+      <form ng-submit="say()">
```

Comment:
Not sure why you're calling say() here. Also, none of the buttons are submitting the form, so I don't think this is even running.

The way I would set this up is call finalize() here in the ng-submit, and remove the ng-click on the finalize button.